### ISSUE TITLE
Auto-email users information about their reserved, checked out, and overdue items

### ISSUE THREAD

**Issue by CalTeach** [joshlam/calteach#198]
Users would receive an email immediately upon reserving an item and checking out an item. Additionally, users would receive a reminder email one day before their item due date and a notice email one day after their item due date, copying a master email as well.

**Comment by joshlam** [comment]
will take a look at this sometime over the weekend

**Comment by joshlam** [comment]
Can you provide some email templates for the reserving/checkout emails?

**Comment by CalTeach** [comment]
Of course. I've emailed all templates to your Zendesk address along with the email to use as both the BCC and the reply-to. Would you please also make the following changes?

* Remove ability for non-admin users to adjust the "Date In" value of their reserved and checked out items
* Change the redirect page for non-logged-in visitors that go to ["My Reservations"](http://calteach.herokuapp.com/reservations) from [this](http://calteach.herokuapp.com/items) to a log-in page that redirects back to the "My Reservations" page post-login
* Add ability to specify "Date In" and "Date Out" when admins click on "Checkout item" for any item

**Comment by joshlam** [comment]
Sure

**Comment by CalTeach** [comment]
Just following up to find out your progress on this.

**Comment by joshlam** [comment]
Apologies, I have not been able to commit that much time to this in the midst of work and travel.

Can you prioritize the four features in terms of urgency/the ones that you would like to have soonest?

>* Remove ability for non-admin users to adjust the "Date In" value of their reserved and checked out items
>* Change the redirect page for non-logged-in visitors that go to ["My Reservations"](http://calteach.herokuapp.com/reservations) from [this](http://calteach.herokuapp.com/items) to a log-in page that redirects back to the "My Reservations" page post-login
>* Add ability to specify "Date In" and "Date Out" when admins click on "Checkout item" for any item

>Users would receive an email immediately upon reserving an item and checking out an item. Additionally, users would receive a reminder email one day before their item due date and a notice email one day after their item due date, copying a master email as well.

**Comment by CalTeach** [comment]
No worries. I've prioritized the requests below:

1. Remove ability for non-admin users to adjust the "Date In" value of their reserved and checked out items **(urgent)**

2. Add ability to specify "Date In" and "Date Out" when admins click on "Checkout item" for any item **(desired now; urgent after 3 is implemented)**

3. Users would receive an email immediately upon reserving an item and checking out an item. Additionally, users would receive a reminder email one day before their item due date and a notice email one day after their item due date, copying a master email as well. **(desired before July 2015)**

4. Change the redirect page for non-logged-in visitors that go to ["My Reservations"](http://calteach.herokuapp.com/reservations) from [this](http://calteach.herokuapp.com/items) to a log-in page that redirects back to the "My Reservations" page post-login **(urgent after 3 is implemented)**

**Comment by aatash** [comment]
Hi! I'm working on #1 (Remove ability for non-admin users to adjust the "Date In" value of their reserved and checked out items) - and I have a question:

I have a solution that removes the ability of a non-admin user to adjust both the "Date In" AND the "Date Out" for their existing reservations. Reasons are:
- If the "Date In" is fixed (not editable) but the "Date out" is not fixed, then the user might try to adjust the "Date Out" to conflict with the "Date In" and cause confusion.
- If the user changes the "Date Out"...it will most likely mean there will be a different "Date In" date. Like if the new "Date Out" is week later than the original, chances are they the "Date In" will want to be sometime a week later too.
- It seems like a better user experience for the whole reservation to be uneditable as opposed to only the start date. Then if they wanted to adjust it from scratch, they can just delete the reservation and start a new one.

What do you think? Let me know if I'm not understanding this fully

**Comment by CalTeach** [comment]
Thanks for contributing to this! No type of user should have the ability to adjust the "Date Out" field of their reservations and non-admin users should not have the ability to adjust the "Date In" field of "Checked Out" items. I hope this helps.

**Comment by joshlam** [comment]
Working on 3 right now, but changes for 2 and 4 should be up. Not quite sure what the status of 1 is? @aatash

**Comment by joshlam** [comment]
Changes for 3 are up.

**Comment by joshlam** [comment]
![screen shot 2015-06-06 at 11 18 50 am](https://cloud.githubusercontent.com/assets/5393387/8020936/10fb12b2-0c41-11e5-92d6-6728729a9d65.png)
![screen shot 2015-06-06 at 11 00 19 am](https://cloud.githubusercontent.com/assets/5393387/8020935/10ed8430-0c41-11e5-8fe2-49e5d2cb585d.png)

**Comment by CalTeach** [comment]
This looks amazing! After testing it just now, it appears that that the printed reservation quantity in the automated email is listing the total quantity of the item rather than the reserved quantity of the item from the inventory. I haven't had any other issues with item 3 so far though but will keep you updated. Thanks again for your hard work, we truly appreciate it!

**Comment by joshlam** [comment]
Fix for listing reserved quantity instead of total quantity is up.

**Comment by CalTeach** [comment]
I haven't received a "check-out is due" or "check-out is now overdue" email for the test check-out that I processed from a test-reservation last week. Any ideas?

**Comment by joshlam** [comment]
The scheduler that runs the reminder might not be running. I will try to take a look at it later in the week

**Comment by joshlam** [comment]
The scheduler was calling `overdue_reminder` instead of `overdue_reminders`, should be working now

**Comment by CalTeach** [comment]
Appears to be working now. Thanks for catching that.

**Comment by CalTeach** [comment]
A student has reported receiving daily emails for a non-existent overdue checkout. Any help would be appreciated, I can give more details if you need them.

**Comment by joshlam** [comment]
Can you provide the email of the student? I can use it to lookup the reservation in the database

**Comment by joshlam** [comment]
The reminder tasks didn't account for canceled reservations, so I suspect that to be the problem. I've pushed up changes for that.

**Comment by CalTeach** [comment]
Thanks again for your help. Do you still want the student's email address?

**Comment by joshlam** [comment]
Was the email address any of the following? eugeniat94@berkeley.edu, k.hosoya@berkeley.edu, r.b.sibal@berkeley.edu, lbwalder@berkeley.edu, pkly@berkeley.edu, jpadilla94@berkeley.edu

**Comment by CalTeach** [comment]
It was, yes.

**Comment by CalTeach** [comment]
When you're able, please change the originating email for the automated emails from calteach@berkeley.edu to calteach_peers@berkeley.edu.

**Comment by joshlam** [comment]
> When you're able, please change the originating email for the automated emails from calteach@berkeley.edu to calteach_peers@berkeley.edu.

Pushed up a change for that, should take effect in a few minutes
